src/software/field_tests/field_test_fixture.py

```diff
@@ -0,0 +1,411 @@
+
+        """
+
+        super(FieldTestRunner, self).__init__(
+            test_name,
+            blue_full_system_proto_unix_io,
+            yellow_full_system_proto_unix_io,
+            gamecontroller,
+        )
+        self.publish_validation_protos = publish_validation_protos
+
+        logger.info("determining robots on field")
+        # survey field for available robot ids
+        try:
+            world = self.world_buffer.get(block=True, timeout=WORLD_BUFFER_TIMEOUT)
+            logger.info(world)
+            self.initial_world = world
+            self.friendly_robot_ids_field = [
+                robot.id for robot in world.friendly_team.team_robots
+            ]
+            self.enemy_robot_ids_field = [
+                robot.id for robot in world.enemy_team.team_robots
+            ]
+
+            logger.info(f"blue team ids {self.friendly_robot_ids_field}")
+
+            if len(self.friendly_robot_ids_field) == 0:
+                raise Exception("no friendly robots found on field")
+
+        except queue.Empty as empty:
+            raise Exception("unable to determine robots on the field")
+
+    def set_tactics(
+        self, tactics: AssignedTacticPlayControlParams, isBlue,
+    ):
+        if isBlue:
+            self.blue_full_system_proto_unix_io.send_proto(
+                AssignedTacticPlayControlParams, tactics
+            )
+        else:
+            self.yellow_full_system_proto_unix_io.send_proto(
+                AssignedTacticPlayControlParams, tactics
+            )
+
+    def set_play(self, play: Play, isBlue):
+        if team == proto.ssl_gc_common_pb2.Team.BLUE:
+            self.blue_full_system_proto_unix_io.send_proto(Play, play)
+
+        else:
+            self.yellow_full_system_proto_unix_io.send_proto(Play, play)
+
+    def send_gamecontroller_command(
+        self,
+        gc_command: proto.ssl_gc_state_pb2.Command,
+        team: proto.ssl_gc_common_pb2.Team,
+        final_ball_placement_point=None,
+    ):
+
+        self.gamecontroller.send_ci_input(
+            gc_command=gc_command,
+            team=team,
+            final_ball_placement_point=final_ball_placement_point,
+        )
+
+    def set_worldState(self, worldstate: WorldState):
+        def __validate(eventually_validation_set, timeout):
+            timeout_time = time.time() + timeout
+
+            while time.time() < timeout_time:
+                try:
+                    current_world = self.world_buffer.get(
+                        block=True, timeout=WORLD_BUFFER_TIMEOUT
+                    )
+
+                    (
+                        eventually_validation_status,
+                        always_validation_status,
+                    ) = validation.run_validation_sequence_sets(
+                        world=current_world,
+                        eventually_validation_sequence_set=[
+                            [eventually_validation_set]
+                        ],
```

Comment:
the readability of this function could be improved if we added some helpers for it. something like:

`
def set_worldState(self, worldstate: WorldState):
    self.validateRobotIds(worldstate);
    self.set_ballState(worldstate);
    self.set_robotInitialState(worldstate);
`